tests/FileStorageS3Test.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+
+namespace Keboola\ManageApiTest;
+
+use Keboola\ManageApi\ClientException;
+use function GuzzleHttp\json_encode;
+
+/**
+ * @group FileStorage
+ */
+class FileStorageS3Test extends ClientTestCase
+{
+
+    private const ROTATE_ABS_OPTIONS = [
+        'accountKey' => TEST_ABS_ROTATE_ACCOUNT_KEY,
+    ];
+
+    private const DEFAULT_S3_OPTIONS = [
+        'awsKey' => TEST_S3_KEY,
+        'awsSecret' => TEST_S3_SECRET,
+        'region' => TEST_S3_REGION,
+        'filesBucket' => TEST_S3_FILES_BUCKET,
+        'owner' => 'keboola',
+    ];
+
+
+    private const ROTATE_S3_OPTIONS = [
+        'awsKey' => TEST_S3_ROTATE_KEY,
+        'awsSecret' => TEST_S3_ROTATE_SECRET,
+    ];
+
+    public function testFileStorageS3Create()
+    {
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $this->assertArrayNotHasKey('awsSecret', $storage);
+        $this->assertSame($storage['awsKey'], TEST_S3_KEY);
+        $this->assertSame($storage['region'], TEST_S3_REGION);
+        $this->assertSame($storage['filesBucket'], TEST_S3_FILES_BUCKET);
+        $this->assertSame($storage['provider'], 'aws');
+        $this->assertFalse($storage['isDefault']);
+    }
+
+    public function testRotateS3Key()
+    {
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $rotatedStorage = $this->client->rotateS3FileStorageCredentials($storage['id'], self::ROTATE_S3_OPTIONS);
+
+        $this->assertArrayNotHasKey('awsSecret', $storage);
+        $this->assertSame($rotatedStorage['awsKey'], TEST_S3_ROTATE_KEY);
+        $this->assertSame($rotatedStorage['region'], TEST_S3_REGION);
+        $this->assertSame($rotatedStorage['filesBucket'], TEST_S3_FILES_BUCKET);
+        $this->assertSame($rotatedStorage['provider'], 'aws');
+        $this->assertFalse($rotatedStorage['isDefault']);
+    }
+
+    public function testListS3Storages()
+    {
+        $initCount = count($this->client->listS3FileStorage());
+        $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+        $storages = $this->client->listS3FileStorage();
+
+        $this->assertSame($initCount + 1, count($storages));
+
+        foreach ($storages as $storage) {
+            if ($storage['provider'] !== 'aws') {
+               $this->fail('List of S3 storages contains also Azure Blob Storages');
+            }
+        }
+    }
+
+    public function testSetS3StorageAsDefault()
+    {
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $this->assertFalse($storage['isDefault']);
+
+        $storage = $this->client->setS3FileStorageAsDefault($storage['id']);
+
+        $this->assertTrue($storage['isDefault']);
+
+        $storageList = $this->client->listS3FileStorage();
+        foreach ($storageList as $item) {
+            if ($item['isDefault'] && $item['id'] !== $storage['id']) {
```

Comment:
Default je vázaný na region storage, může jich být víc.

Reply:
takže bych přidal ještě kontrolu na region.

Reply:
Přidal jsem test a přidal storage do 2 regionů u S3. Vyžaduje to funkční credentials na jiný region, takže jsem ten test dal skipped a napsal, že musíš růčo doplnit parametry a nepřidával jsem to .env, protože ten test se normálně stejně nespouští.